test/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs

```diff
@@ -0,0 +1,510 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.ProjectManagement;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using Test.Utility;
+using Xunit;
+
+namespace NuGet.CommandLine.Test
+{
+    public class NuGetUpdateCommandTests
+    {
+        [Fact]
+        public async Task UpdateCommand_Success_References()
+        {
+            var packagesSourceDirectory = TestFilesystemUtility.CreateRandomTestFolder();
```

Comment:
There is an opportunity to factor this code into helper methods or helper private test class. Will try to do that if time permits